compiler/rustc_resolve/src/errors.rs

```diff
@@ -0,0 +1,279 @@
+use rustc_macros::{SessionDiagnostic, SessionSubdiagnostic};
+use rustc_span::{
+    symbol::{Ident, Symbol},
+    Span,
+};
+
+#[derive(SessionDiagnostic)]
+#[diag(resolve::parent_module_reset_for_binding, code = "E0637")]
+pub(crate) struct ParentModuleResetForBinding;
+
+#[derive(SessionDiagnostic)]
+#[note]
+#[diag(resolve::ampersand_used_without_explicit_lifetime_name, code = "E0637")]
+pub(crate) struct AmpersandUsedWithoutExplicitLifetimeName(#[primary_span] pub(crate) Span);
+
+#[derive(SessionDiagnostic)]
+#[note]
+#[diag(resolve::underscore_lifetime_name_cannot_be_used_here, code = "E0637")]
+pub(crate) struct UnderscoreLifetimeNameCannotBeUsedHere(#[primary_span] pub(crate) Span);
+
+#[derive(SessionDiagnostic)]
+#[diag(resolve::crate_may_not_be_imported)]
+pub(crate) struct CrateMayNotBeImprted(#[primary_span] pub(crate) Span);
+
+#[derive(SessionDiagnostic)]
+#[diag(resolve::crate_root_imports_must_be_named_explicitly)]
+pub(crate) struct CrateRootNamesMustBeNamedExplicitly(#[primary_span] pub(crate) Span);
+
+#[derive(SessionDiagnostic)]
+#[diag(resolve::crate_root_imports_must_be_named_explicitly)]
+pub(crate) struct ResolutionError(#[primary_span] pub(crate) Span);
+
+#[derive(SessionDiagnostic)]
+#[diag(resolve::name_is_already_used_as_generic_parameter, code = "E0403")]
+pub(crate) struct NameAlreadyUsedInParameterList {
+    #[primary_span]
+    #[label]
+    pub(crate) span: Span,
+
```

Comment:
We can remove the newlines between fields :)

Reply:
sure, thing.